src/ui/public/vislib/lib/handler/handler.js

```diff
@@ -203,13 +204,19 @@ define(function (require) {
      * @method destroy
      */
     Handler.prototype.destroy = function () {
-      this.charts.forEach(function (chart) {
+      this.binder.destroy();
+
+      this.renderArray.forEach(function (renderable) {
+        if (_.isFunction(renderable.destroy)) {
+          renderable.destroy();
+        }
+      });
+
+      this.charts.splice(0).forEach(function (chart) {
```

Comment:
The `splice()` is to empty the previous array before iterating. This way `this.charts` is treated like a queue which can be emptied many times. It's mostly just a habbit